### ISSUE TITLE
numpy.unique of empty array with axis kwarg errors

### ISSUE THREAD

**Issue by Helveg** [numpy/numpy#18444]
### Reproducing code example:

```python
import numpy
numpy.unique(numpy.empty((0, 2)), axis=0)
```

### Error message:

```
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "<__array_function__ internals>", line 5, in unique
  File "/opt/python/3.8.2.1/lib/python3.8/site-packages/numpy/lib/arraysetops.py", line 275, in unique
    ar = ar.reshape(orig_shape[0], -1)
ValueError: cannot reshape array of size 0 into shape (0,newaxis)
```

### NumPy/Python version information:

```
1.18.2 3.8.2 (default, May 13 2020, 11:50:46)
[GCC 8.1.0 20180502 (Cray Inc.)]
```

**Comment by eric-wieser** [comment]
Right, I guess the `-1` needs to be replaced with `functools.reduce(operator.prod, orig_shape[1:], 1])`

**Comment by eric-wieser** [comment]
Closed in gh-15747 (1.19.0rc1 onwards), duplicate of gh-15559.

**Comment by eric-wieser** [comment]
Duplicate of #15559
